### PR TITLE
[Feature] Save column order in data store

### PR DESCRIPTION
**Pull request by SferaDev**
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/1xwfebr

### :memo: Implementation

Added usecases and methods for: saving selected columns on the data store, and for retrieving them when component renders.

### CODE REVIEW

**Comment by SferaDev** [comment]
Thanks @p3rcypj, @idelcano will functionally review the feature

File: src/webapp/components/user-list-table/UserListTable.tsx [hunk 1]
```diff
@@ -222,20 +240,44 @@ export const UserListTable: React.FC<UserListTableProps> = props => {
...
+
+    useEffect(() => {
+        compositionRoot.users.getColumns().run(
+            columns => {
+                setVisibleColumns(columns);
+                onChangeVisibleColumns(columns);
+            },
+            error => snackbar.error(error)
+        );
+    }, [compositionRoot, snackbar, onChangeVisibleColumns]);
```
Comment (SferaDev) [inline src/webapp/components/user-list-table/UserListTable.tsx:272]:
```suggestion
    useEffect(() => {
        return compositionRoot.users.getColumns().run(
            columns => {
                setVisibleColumns(columns);
                onChangeVisibleColumns(columns);
            },
            error => snackbar.error(error)
        );
    }, [compositionRoot, snackbar, onChangeVisibleColumns]);
```

**Comment by idelcano** [comment]
The feature looks great.
But could we add some default columns?
When i open the app the first time looks strange:
screenshot-localhost_8081-2022 02 09-10_47_58


And when add more columns the header looks uggly too:

screenshot-localhost_8081-2022 02 09-10_49_26

**Comment by SferaDev** [comment]
> The feature looks great. But could we add some default columns? When i open the app the first time looks strange: screenshot-localhost_8081-2022 02 09-10_47_58

Indeed, @p3rcypj can you take a look?

> And when add more columns the header looks uggly too:

We will not fix this, happens on all our apps

**Comment by p3rcypj** [comment]
Sure. Should I add the initial columns if there is no column selected?

**Comment by SferaDev** [comment]
> Sure. Should I add the initial columns if there is no column selected?

No, it is more for the user when it enters the first time that the columns will be empty. You can force that by creating a new user and trying to enter the app @p3rcypj

**Comment by SferaDev** [comment]
Thanks @p3rcypj

**Pull request merged**